### ISSUE TITLE
As a user I can automatically sync my data with a cloud service

### ISSUE THREAD

**Issue by flauschtrud** [flauschtrud/broccoli#171]
Google Drive, Nextcloud, Owncloud (Request)

**Comment by wbrtm** [comment]
First of all: this is a great app, thank you for sharing!
First thing I figured was missing was any way of syncing. Having your recipies stored in an accessible file would be a first step towards this feature. Users then could manually sync that file. Bonus points for individual files per recipe to be able to sync with other devices and people, e. g. using Syncthing.

**Comment by flauschtrud** [comment]
> First of all: this is a great app, thank you for sharing! One thing I figured was missing was some way of syncing. Having your recipies stored in an accessible file would be a first step towards this feature, I could not find where they are stored currently (Android). Users then could manually sync that file. Bonus points for individual files per recipe to be able to sync with other devices and people, e. g. using Syncthing.

A manual backup is possible already via the export feature (backup & restore in the menu), which basically creates a zip archive which can be manually backed up or shared. Of course this is not a sync but just a backup, but it helps to protect the data.

You can also create export files of individual recipes (recipe menu -> share as file), these files can also be shared with other people and reimported. Again, nothing fancy.

(please note, restoring backup files creates duplicate recipes in case you already have any in your database)

**Comment by wbrtm** [comment]
Thank you for your fast reply. I already played with the backup and sharing functions and they work as intended. What I am missing is a way of sharing new recipes and syncing changes between devices. The way recipes are stored as of now prevents doing that with another app like Syncthing or any cloud app. What I wanted to suggest here is that it does not need an elaborate integration for cloud services but having recipes stored in an accessible location in e.g. json format would allow users to sync with their service of choice. As I am not familiar with mobile development I am not sure this is as straightforward as it sounds though.

**Comment by flauschtrud** [comment]
Yes, the recipes (plus categories and search indices) are stored in an SQLite database, so this won't be feasible for a sync with something like Syncthing. Only the images are stored in the app's storage.

**Comment by rubikscuber** [comment]
The existing import/export functionality is quite good. A nice to have here, would be to have a checkbox to exclude exact duplicates (e.g. all fields are the same) from the import. The possibility to delete multiple recipes at once (by first selecting them in the overview) could be an useful alternative.

**Comment by flauschtrud** [comment]
> The existing import/export functionality is quite good. A nice to have here, would be to have a checkbox to exclude exact duplicates (e.g. all fields are the same) from the import. The possibility to delete multiple recipes at once (by first selecting them in the overview) could be an useful alternative.

Yes, both very valid requests. I created a new issue for the import #212 and the multi deletion already has one #143

**Comment by meersjo** [comment]
In the interest of freedom from commercial services, please consider adding a few more traditional services like FTP, SCP, Samba etc as well :-)
